lib/s3_writer.rb

```diff
@@ -41,7 +51,9 @@ def feed_xml(checkouts)
           xml.entry {
             xml.id "#{checkout.id}-#{checkout.barcode}"
             title = "\"#{checkout.title}\""
-            title += " by #{checkout.author}" if checkout.has? :author
+            checkout_author = checkout.has? :author ? get_author(checkout.author) : ""
```

Comment:
I believe this is the source of the test failure. Ternary operator is evaluated first before sending to `:has`. Without parens, this is interpreted as:

```
checkout_author = checkout.has?(:author ? get_author(checkout.author) : "")
```

Because  a symbol like `:author` is always truthy regardless of value, the first part of the ternary is evaluated, causing this to be called, essentially:

```
checkout_author = checkout.has?(" by Cee-Lo (Musician)")
```

Should add parens to make the precedence explicit as follows:

```
checkout_author = checkout.has?(:author) ? get_author(checkout.author) : ""
```

Reply:
ahhhh interesting! So the `?` would be interpreted as the place to decide the condition.

Reply:
Updated! All the tests now pass. Thank you!